server/src/error.rs

```diff
@@ -20,20 +52,54 @@ impl IntoResponse for CheeseError {
     type BodyError = <Self::Body as axum::body::HttpBody>::Error;
 
     fn into_response(self) -> axum::http::Response<Self::Body> {
-        let bb = match self {
-            Self::Mongo(e) => {
-                event!(Level::ERROR, "MongoDB error: {:?}", e);
-                Self::Body::from("MongoDB did a fuckywucky")
-            }
-            Self::Oid(e) => {
-                event!(Level::ERROR, "ObjectID error: {:?}", e);
-                Self::Body::from("ObjectID did a fuckywucky")
-            }
-        };
-
-        Response::builder()
-            .status(StatusCode::INTERNAL_SERVER_ERROR)
-            .body(bb)
-            .unwrap()
+        match self {
+            Self::Mongo(e) => response_error!(e, internal server error),
+            Self::Oid(e) => response_error!(e, internal server error),
+            CheeseError::Login(l) => match l {
+                e @ LoginError::InvalidCredentials => {
+                    response_error!(e, "unauthorized", StatusCode::UNAUTHORIZED)
+                }
+                LoginError::Database(e) => response_error!(e, internal server error),
+                LoginError::Bcrypt(e) => response_error!(e, internal server error),
+                e @ LoginError::UserWithoutId => response_error!(e, internal server error),
+            },
+            CheeseError::Register(r) => match r {
+                RegisterError::Database(e) => response_error!(e, internal server error),
+                RegisterError::Bcrypt(e) => response_error!(e, internal server error),
+                e @ RegisterError::UserExists => {
+                    response_error!(e, "user exists", StatusCode::CONFLICT)
+                }
+            },
+            CheeseError::Authorization(a) => match a {
```

Comment:
A bit of inconsistent formatting here, sometimes you return a `{ response_error!(..) }` and sometimes `response_error!(..)` directly.

Reply:
cargo format does this